source/CIS/cis-3.rst

```diff
@@ -0,0 +1,144 @@
+
+Logged events
+-------------
+
+This standard does not introduce any additional events. Entrypoints invoked via the `permit` function using the sponsored
+transaction mechanism MUST emit the same events as if they were invoked
+by the sponsoree directly without the sponsored transaction mechanism.
+
+Contract function
+-----------------
+
+A smart contract implementing CIS-3 MUST export the functions: :ref:`CIS-3-functions-permit` and :ref:`CIS-3-functions-supportsPermit`.
+
+.. _CIS-3-functions-permit:
+
+``permit``
+^^^^^^^^^^
+
+Verifies an ed25519 signature from a sponsoree and authorizes the sponsor to execute the logic of
+specific entrypoints on behalf of the sponsoree. The sponsored transaction mechanism replaces the
+authorization checks conducted on the `sender/invoker` variable with signature verification.
+
+Parameter
+~~~~~~~~~
+
+The parameter (``PermitParam``) contains a two-level signature map and a signer account that created the signature
+together with the message that was signed.
+
+.. note::
+
+    The CIS3 standard supports multi-sig accounts which is the purpose of the two-level signature map. A basic account (no multi-sig account) SHOULD have its signature at the key 0 in both maps.
+
+The message (``PermitMessage``) contains a contract_address (``ContractAddress``), entry_point (``OwnedEntrypointName``), nonce (``u64``), timestamp (``Timestamp``), and the payload (``PermitPayload``).
+This message structure enables the sponsoree the authorize the sponsor to act on its behalf in the given scope.
+
+The payload (``PermitPayload``) is serialized as: The first byte to distinguish the entrypoint followed by the parameter intended for that entrypoint::
+
+  PermitPayload  ::= (0: Byte) (Entrypoint_1_Parameter) // First entrypoint supported
+                   | (1: Byte) (Entrypoint_2_Parameter) // Second entrypoint supported
+                   | (2: Byte) (Entrypoint_3_Parameter) // Third entrypoint supported
+                   ...
+
+  PermitMessage ::= (contract_address: ContractAddress) (entry_point: OwnedEntrypointName) (nonce: u64) (timestamp: Timestamp) (payload: PermitPayload)
+
+  PermitParam ::= (signature: BTreeMap<u8, BTreeMap<u8, SignatureEd25519>>) (signer: AccountAddress) (message: PermitMessage)
+
+.. note::
+
+    The PermitPayload is an enum and the CIS-3 standard specifies that the first byte is used to distinguish between the different entrypoints.
+    This definition enables the serialization of 255 different entrypoints which should be sufficient in practice.
+
+Requirements
+~~~~~~~~~~~~
+
+- The requirements specified for an entrypoint MUST persist even if invoked via the `permit` function. E.g. a smart contract implementing an `updateOperator/transfer` function from the CIS-2 standard, if these entrypoints are invoked via the `permit` function, the sponsored transaction invoke MUST adhere to the CIS-2 standard as well.
```

Comment:
A stronger requirement (unrelated to the nonce) could be that the entry point MUST result in the same outcome as if it was invoked directly by the sponsoree.
This should "prevent" changing behavior when the transaction is sponsored.